tests_new/integration_tests/modules/share_base/test_new_crossacc_s3_share.py

```diff
@@ -0,0 +1,393 @@
+):
+    share = create_share_object(
+        client=client5,
+        dataset_or_item_params={'datasetUri': session_s3_dataset1.datasetUri},
+        environmentUri=persistent_cross_acc_env_1.environmentUri,
+        groupUri=group5,
+        principalId=group5 if principal_type == 'Group' else consumption_role_1.consumptionRoleUri,
+        principalType=principal_type,
+        requestPurpose='test create share object',
+        attachMissingPolicies=True,
+        permissions=['Read'],
+    )
+    share = get_share_object(client5, share.shareUri)
+
+    items = share['items'].nodes
+    assert_that(len(items)).is_greater_than(0)
+    assert_that(items[0].status).is_none()
+
+    item_to_add = items[0]
+    share_item_uri = add_share_item(client5, share.shareUri, item_to_add.itemUri, item_to_add.itemType)
+    assert_that(share_item_uri).is_not_none()
+
+    updated_share = get_share_object(client5, share.shareUri, {'isShared': True})
+    items = updated_share['items'].nodes
+    assert_that(items).is_length(1)
+    assert_that(items[0].shareItemUri).is_equal_to(share_item_uri)
+    assert_that(items[0].status).is_equal_to(ShareItemStatus.PendingApproval.value)
+
+    clean_up_share(client5, share.shareUri)
+
+
+@pytest.mark.parametrize(
+    'principal_type',
+    ['Group', 'ConsumptionRole'],
+)
+def test_reject_share(
+    client1, client5, persistent_cross_acc_env_1, session_s3_dataset1, group5, consumption_role_1, principal_type
+):
+    share = create_share_object(
+        client=client5,
+        dataset_or_item_params={'datasetUri': session_s3_dataset1.datasetUri},
+        environmentUri=persistent_cross_acc_env_1.environmentUri,
+        groupUri=group5,
+        principalId=group5 if principal_type == 'Group' else consumption_role_1.consumptionRoleUri,
+        principalType=principal_type,
+        requestPurpose='test create share object',
+        attachMissingPolicies=True,
+        permissions=['Read'],
+    )
+    share = get_share_object(client5, share.shareUri)
+
+    items = share['items'].nodes
+    assert_that(len(items)).is_greater_than(0)
+    assert_that(items[0].status).is_none()
+
+    item_to_add = items[0]
+    add_share_item(client5, share.shareUri, item_to_add.itemUri, item_to_add.itemType)
+    submit_share_object(client5, share.shareUri)
+
+    reject_share_object(client1, share.shareUri)
+    updated_share = get_share_object(client1, share.shareUri)
+    assert_that(updated_share.status).is_equal_to(ShareObjectStatus.Rejected.value)
+
+    change_request_purpose = update_share_reject_reason(client1, share.shareUri, 'new purpose')
+    assert_that(change_request_purpose).is_true()
+    updated_share = get_share_object(client5, share.shareUri)
+    assert_that(updated_share.rejectPurpose).is_equal_to('new purpose')
+
+    clean_up_share(client5, share.shareUri)
+
+
+@pytest.mark.parametrize(
+    'share_fixture_name',
+    ['session_share_1', 'session_share_consrole_1'],
+)
```

Comment:
Similar to my previous comment do you think it's possible to use parametrized fixtures here as well to avoid the `getfixturevalue`?

Reply:
implemented